teachertool/src/components/styling/shared.scss

```diff
@@ -0,0 +1,261 @@
+/**
+ * This file contains all the shared styles for application components. Do not
+ * import this file directly into components. Instead, import into the
+ * component's module.scss file so that it will be scoped to that module.
+ *
+ * Classes in this file are composable, modeled after [Tailwind CSS](https://tailwindcss.com)

```

Comment:
I love tailwind's composable classes specifically for controlling flex layout. I was building my own set and it was starting to look a lot like tailwind, so I just copied this tiny subset from tailwind. It doesn't imply we should adopt tailwind wholesale.

Reply:
Yeah I'm a little skeptical about this re-inventing tailwind from scratch setup, personally. We're getting into a weird hybrid state where some components are styled with more typical css classes and some use these composable ones, and I think it's likely to get messy and unclear which approach we should use when...

Reply:
I don't think we should adopt tailwind. It has a lot of functionality I think we wouldn't want to make available to our codebase, like color palettes and media breakpoints. We already have these. I'd rather take inspiration from it where it makes sense, in limited ways.

Reply:
Hmm, but even with the classes here, what's the rule of thumb for when you should make a specific css class for a component vs grouping together a bunch of these?

Reply:
I guess I don't see what defined "where it makes sense"

Reply:
For example, I'm not sure why the ProjectWorkspace root would use "panel", "h-full", "w-full", "flex", "flex-col" instead of having its own class.

I actually quite like tailwind myself, so I understand the desire to have that functionality, but I'm concerned about what this in-between approach will end up looking like

Reply:
I'd say rule of thumb is: don't use these classes unless it seems more natural to do so. They can be useful to inject additional css into a react-common component by className prop. I may end up not using them where I am now.

Reply:
> I guess I don't see what defined "where it makes sense"

where it makes sense = where functionality wouldn't conflict with existing systems. color theming and media breakpoints being two examples where conflict would occur.

Reply:
> For example, I'm not sure why the ProjectWorkspace root would use "panel", "h-full", "w-full", "flex", "flex-col" instead of having its own class.
> 
> I actually quite like tailwind myself, so I understand the desire to have that functionality, but I'm concerned about what this in-between approach will end up looking like

That wasn't the best use of these classes, I admit. That code evolved over time, and should have evolved once more to move flex functionality into the panel class.

Reply:
If these classes are a sticking point, I can remove them. They aren't colliding with anything, but they're also just a convenience layer.